components/arm/client.go

```diff
@@ -200,24 +198,5 @@ func (c *client) updateKinematics(ctx context.Context, extra map[string]interfac
 		return nil, err
 	}
 
-	format := resp.GetFormat()
-	data := resp.GetKinematicsData()
-
-	switch format {
-	case commonpb.KinematicsFileFormat_KINEMATICS_FILE_FORMAT_SVA:
-		return referenceframe.UnmarshalModelJSON(data, c.name)
-	case commonpb.KinematicsFileFormat_KINEMATICS_FILE_FORMAT_URDF:
-		modelconf, err := urdf.UnmarshalModelXML(data, c.name)
-		if err != nil {
-			return nil, err
-		}
-		return modelconf.ParseConfig(c.name)
-	case commonpb.KinematicsFileFormat_KINEMATICS_FILE_FORMAT_UNSPECIFIED:
-		fallthrough
-	default:
-		if formatName, ok := commonpb.KinematicsFileFormat_name[int32(format)]; ok {
-			return nil, fmt.Errorf("unable to parse file of type %s", formatName)
-		}
-		return nil, fmt.Errorf("unable to parse unknown file type %d", format)
-	}
+	return updateKinematicHelper(c.name, resp)
```

Comment:
This whole function was already meant to be the helper.  You can just use this function in both places if you change the arguments that it takes

Reply:
I am confused by this comment. 

> This whole function was already meant to be the helper.

Do you mean that `(c *client) updateKinematics{}()` itself should be viewed entirely as a helper which is called from server.go?

Reply:
I wanted to keep this from being multiple functions but actually I think its fine and my confusion was due to the naming.  Lets just change the name of this function to something that actually describes whats happening like `parseKinematicsResponse`